### DISCUSSION TITLE
Increase upload file size limit & allow updating existing emotes

### DISCUSSION THREAD

**Discussion by Jnosh** [night/betterttv#6949] [Ideas]
It would be great if the file size limit could be increased a bit. 1MB is fairly strict when uploading 3x emotes.

At least for animated WEBP emotes, BetterTTV recompresses the emotes anyway. I often need to lower the quality of the 3x source image to fit into the 1MB upload limit only to have BetterTTV recompress the emote and the resulting final 3x image to be 2-3MB in size as it then uses a higher quality target when recompressing which is a bit silly.

If the upload limit were a bit higher this would allow submitting higher quality source images. Both 7TV and FFZ have a significantly higher limit (7MB for 7TV for example) but just even a 3MB would help enormously.

If this were implemented, it would also be nice to have the ability to replace the images for existing emotes (or add new versions similar to 7TV) so emotes could be updated with higher quality source images where that makes sense. Currently one would need to delete the original emote and re-upload it, removing the emote for any current users.

**Comment by night** [comment]
can you provide some example originals and targets which show an increase in file size, with exception to a conversion from PNG to WEBP or WEBP to PNG?

**Comment by Jnosh** [comment]
Sure, [here are some examples](https://github.com/user-attachments/files/16322340/bttv_size_increase.zip). The files ending in `_orig` are the originals that were uploaded, the ones ending in `_3x` the ones served on betterttv.com

Edit:
To add some additional context, I generally use ezgif.com for the final assembly which uses webpmux to create the webp files. Generally a quality setting in the low 90s is sufficient to get below 1MB even for somewhat complex emotes but I would still prefer to submit files with as little compression as possible if BTTV is going to recompress them again anyway. At max quality (100) the files are usually in the 1-4MB range for complex emotes.
Today I had an emote that is 3.4MB at quality 100 but really struggled to get below 1MB. I had to drop the quality down to 20 at which point the image quality very noticeably dropped. The emote ended up being 2.9MB after recompression.
